op-e2e/actions/proofs/l1_blob_parameter_forks_test.go

```diff
@@ -0,0 +1,141 @@
+				require.True(t, isActive(l1Origin.Number, l1Origin.Time), "%s not active at l1 origin %d, time %d", label, l1Origin.Number, l1Origin.Time)
+			} else {
+				require.False(t, isActive(l1Origin.Number, l1Origin.Time), "%s should not be active at l1 origin %d, time %d", label, l1Origin.Number, l1Origin.Time)
+			}
+			bbfL1 := eip4844.CalcBlobFee(env.Sd.L1Cfg.Config, l1Origin)
+
+			require.True(t, bbfL2.Cmp(bbfL1) == 0,
+				"%s: blob base fee does not match, bbfL2=%d, bbfL1=%d, l1BlockNum=%d, l2BlockNum=%d", label, bbfL2, bbfL1, l1Origin.Number, l2Block.Number)
+
+			require.True(t, bbfL2.Cmp(big.NewInt(1)) > 0,
+				"%s: blob base fee is unrealistically low and doesn't exercise the blob fee calculation", label)
+		}
+
+		// buildL1ToTime advances L1 with empty blocks until the given fork time.
+		buildL1ToTime := func(t actionsHelpers.StatefulTesting, forkTime *uint64) {
+			require.NotNil(t, forkTime, "fork time must be configured")
+			h := miner.L1Chain().CurrentHeader()
+			for h.Time < *forkTime {
+				h = miner.ActEmptyBlock(t).Header()
+			}
+		}
+
+		// Iterate through all forks and assert pre/post activation blob fees match expectations
+		cfg := env.Sd.L1Cfg.Config
+		forks := []struct {
+			label    string
+			forkTime *uint64
+			isActive func(num *big.Int, time uint64) bool
+		}{
+			{"Prague", cfg.PragueTime, func(num *big.Int, time uint64) bool { return cfg.IsPrague(num, time) }},
+			{"Osaka", cfg.OsakaTime, func(num *big.Int, time uint64) bool { return cfg.IsOsaka(num, time) }},
+			{"BPO1", cfg.BPO1Time, func(num *big.Int, time uint64) bool { return cfg.IsBPO1(num, time) }},
+			{"BPO2", cfg.BPO2Time, func(num *big.Int, time uint64) bool { return cfg.IsBPO2(num, time) }},
+			{"BPO3", cfg.BPO3Time, func(num *big.Int, time uint64) bool { return cfg.IsBPO3(num, time) }},
+			{"BPO4", cfg.BPO4Time, func(num *big.Int, time uint64) bool { return cfg.IsBPO4(num, time) }},
+		}
+		for _, f := range forks {
+			// Advance L1 to fork activation
+			buildL1ToTime(t, f.forkTime)
+
+			// Build an empty L2 block which still has a pre-fork L1 origin, and check blob fee
+			sequencer.ActL2EmptyBlock(t)
+			l2Block := sequencer.SyncStatus().UnsafeL2
+			requireConsistentBlobBaseFeeForFork(t, l2Block, false, f.label, f.isActive)
+
+			// Advance L2 chain until L1 origin is at/after the fork activation
+			sequencer.ActL1HeadSignal(t)
+			sequencer.ActBuildToL1HeadUnsafe(t)
+
+			l2Block = sequencer.L2Unsafe()
+			require.Greater(t, l2Block.Number, uint64(1))
+			requireConsistentBlobBaseFeeForFork(t, l2Block, true, f.label, f.isActive)
+		}
+
+		// Final sync
+		env.BatchMineAndSync(t)
+
+		// Run fault proof program
+		safeL2Head := sequencer.L2Safe()
+		env.RunFaultProofProgramFromGenesis(t, safeL2Head.Number, testCfg.CheckResult, testCfg.InputParams...)
+	}
+
+	matrix := helpers.NewMatrix[any]()
+	defer matrix.Run(gt)
+	matrix.AddDefaultTestCases(nil, helpers.NewForkMatrix(helpers.LatestFork), runBlobParameterForksTest)
```

Comment:
I don't know who ever came up with this idea of using `defer` to execute the test 😅 it should just be invoked normally. `defer` should only really be used to free up resources (mutex unlocking, context closing, closing of files, etc.), including during panic-induced stack unwinding...
```suggestion
	matrix := helpers.NewMatrix[any]()
	matrix.AddDefaultTestCases(nil, helpers.NewForkMatrix(helpers.LatestFork), runBlobParameterForksTest)
	matrix.Run(gt)
```

Reply:
I do know, but I won't say ;-) Fixed now.